optashift-employee-rostering-gwtui/src/main/java/org/optaplanner/openshift/employeerostering/gwtui/client/pages/rotation/RotationPage.java

```diff
@@ -132,25 +154,29 @@ private void onSaveClicked(final @ForEvent("click") MouseEvent e) {
 
     private void save() {
 
-        final List<ShiftInfo> newShiftInfoList = viewport.getLanes().stream()
-                .flatMap(lane -> lane.getSubLanes().stream())
-                .flatMap(subLane -> subLane.getBlobs().stream())
-                .filter(blob -> blob.getPositionInGridPixels() >= 0) //Removes left-most twins
-                .map(blob -> ((ShiftBlob) blob).getShift())
-                .map(this::newShiftInfo)
-                .collect(toList());
-
-        ShiftRestServiceBuilder.createTemplate(
-                tenantStore.getCurrentTenantId(),
-                newShiftInfoList,
-                onSuccess(i -> refresh()));
+        final List<ShiftTemplate> newShiftInfoList = viewport.getLanes().stream()
+                                                             .flatMap(lane -> lane.getSubLanes().stream())
+                                                             .flatMap(subLane -> subLane.getBlobs().stream())
+                                                             .filter(blob -> {
+                                                                 ShiftBlob shiftBlob = (ShiftBlob) blob;
+                                                                 if (!shiftBlob.getTwin().isPresent()) {
+                                                                     return true;
+                                                                 }
+                                                                 return shiftBlob.getPositionInGridPixels() > shiftBlob.getTwin().get().getPositionInGridPixels();
+                                                             })
```

Comment:
Why change that?

Reply:
I noticed some shifts, those that are on the left boundary, duplicating whenever I click the save button (implying that both the twin and the original were kept). This guarantees only one of them is kept and fixes the issue. Might have changed after roster generation fixes.